### ISSUE TITLE
Multiple screenshots on one page hangs 

### ISSUE THREAD

**Issue by cristiberceanu** [puppeteer/puppeteer#2223]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 1.1.1
* Platform / OS version: Ubuntu server 16.04
* URLs (if applicable):
* Node.js version: v8.9.4

**What steps will reproduce the problem?**
Run a script that tries to take multiple screenshots of the same page without navigating out of the page or refreshing it, just simply trigger 3 screenshots of different areas of the page.

_Please include code that reproduces the issue._

**Sorry for the code formatting in this comment, I don't know why github's markdown doesn't let me add the whole function to the block 

```js
async function takeScreenShots() {
    browser = await puppeteer.launch();
    page = await browser.newPage();
    await page.setViewport({
        width: 1024,
        height: 768
    });

    await page.goto("https://google.com/", {
        waitUntil: ['domcontentloaded'] // ensures images are loaded
    });
    var screenshotParams = [
        {
            "printHeight": 100,
            "printWidth": 100,
            "offsetX": 0,
            "offsetY": 0
        }, {
            "printHeight": 375,
            "printWidth": 473,
            "offsetX": 0,
            "offsetY": 30
        }, {
            "printHeight": 375,
            "printWidth": 473,
            "offsetX": 0,
            "offsetY": 100
        }
    ];
    var count = 0;
    for (let screenshotParam of screenshotsParams) {
        var name = makeid(),
            offsetX = screenshotParam.offsetX,
            offsetY = screenshotParam.offsetY,
            printWidth = screenshotParam.printWidth,
            printHeight = screenshotParam.printHeight;
        // Output a page screenshot
        await page.screenshot({
            path: 'screenshots/' + count + '.png',
            clip: {
                x: offsetX,
                y: offsetY,
                width: printWidth,
                height: printHeight
            }
        });
        count++;
    }
    await page.close();
    await browser.close();
}
```

**What is the expected result?**
I'm expecting that by doing 3 screenshots on the page it'll work. Should also be fairly fast.

**What happens instead?**
by running this function the node process sometimes hangs. The promise does not reject nor resolve, it just seems to stall without producing a screenshot. It always seem to stall on the last item of the array. I.e. if the screenshots had 4 items, it would do first 3 and stop when doing the 4th. It only hangs once every 10 requests. When it works it takes about 3-6 seconds for it to finish.

**Comment by gutnikov** [comment]
Tell us about your environment:

* Puppeteer version: 1.1.1 or 1.2.0
* Platform / OS version: Mac OS Sierra 10.12.2
* Node.js version: v8.9.3

Same. Trying to get multiple screenshots from one page:

```javascript
async function takeScreenshot(page, rect, filePath) {
    await page.screenshot({
        path: filePath,
        clip: {
            x: rect.x,
            y: rect.y,
            width: rect.width,
            height: rect.height
        }
    });
}

async function takeSandboxScreenshots(page, outputDir) {
    const variations = await getSandboxedElements(page);
    log(`Found ${variations.length} sanboxed elements`);
    async function takeNext(current) {
        if (current >= variations.length) {
            return;
        }
        const item = variations[current];
        const filePath = `${outputDir}/${item.component}-${item.variation}.png`;
        await takeScreenshot(page, item.rect, filePath);
        await takeNext(current + 1);
    }
    return takeNext(0);
}
```

It hangs on random element on every script run and in rare cases it doesn't.

**Comment by gutnikov** [comment]
Update:
It seems that i'v managed script to work by setting viewport height to something like 40K pixels, which sounds like a very weird solution )) 40K - is a size of page content that i'm trying to get screenshots from

**Comment by cristiberceanu** [comment]
And it never hangs now?

**Comment by gutnikov** [comment]
Yep, never.
I'm not going to use this anyway ) just trying to figure out what i'v possibly did wrong before.

**Comment by cristiberceanu** [comment]
what was it?

**Comment by gutnikov** [comment]
No, i mean i still don't know why it works with 40K height and how to make it work without setting it

**Comment by cristiberceanu** [comment]
My code seems to randomly hang once every 25-30 executions no matter what parameters I give it

**Comment by darrennolan** [comment]
We're facing the same issue.

* Puppeteer version: 1.3.0
* Platform / OS version: ubuntu:trusty-20180412 (14.04)
* URLs (if applicable): (Single React Storyboard page for visual regressions)
* Node.js version: v8.11.1

**Comment by darrennolan** [comment]
Just adding to this, we did find a workaround for our suite of tests, basically we just refresh the page between tests which 'appeared' to have resolved this particular issue.

We did attempt to set the height to 40k, but that had no luck.

Sadly today I'm having the same issue as before of hanging on screenshots again, even with our refreshes.  Going to try and dig a little deeper and report back with anything I can find out.

Thanks all,
